ui/components/component-library/popover/popover.tsx

```diff
@@ -0,0 +1,133 @@
+    {
+      children,
+      className = '',
+      position = PopoverPosition.Auto,
+      role = PopoverRole.Tooltip,
+      hasArrow = false,
+      matchWidth,
+      preventOverflow = false,
+      offset = [0, 8],
+      flip = false,
+      referenceHidden = true,
+      referenceElement,
+      isOpen,
+      title,
+      isPortal = true,
+      arrowProps,
+      ...props
+    }: PopoverProps,
+    ref: Ref<PopoverRef>,
+  ) => {
+    const [popperElement, setPopperElement] = useState<HTMLElement | null>(
+      null,
+    );
+    const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
+
+    // Define Popper options
+    const { styles, attributes } = usePopper(referenceElement, popperElement, {
+      placement: position,
+      modifiers: [
+        {
+          name: 'preventOverflow',
+          enabled: position === PopoverPosition.Auto ? true : preventOverflow,
+        },
+        {
+          name: 'flip',
+          enabled: position === PopoverPosition.Auto ? true : flip,
+        },
+        {
+          name: 'arrow',
+          enabled: hasArrow,
+          options: {
+            element: arrowElement,
+          },
+        },
+        {
+          name: 'offset',
+          options: {
+            offset,
+          },
+        },
+      ],
+    });
+
+    // Define width to match reference element or auto
+    const contentStyle = {
+      width: matchWidth ? referenceElement?.clientWidth : 'auto',
+    };
+
+    // Forwarding function to close the popover using the ref
+    useImperativeHandle(ref, () => ({
+      closePopover: () => {
+        // Your close popover implementation here
+        console.log('closePopover called');
+      },
+    }));
+
+    const PopoverContent = (
+      <Box
+        borderColor={BorderColor.borderMuted}
+        borderRadius={BorderRadius.LG}
+        backgroundColor={BackgroundColor.backgroundDefault}
+        padding={4}
+        role={role}
+        className={classnames(
+          'mm-popover',
+          {
+            'mm-popover--open': Boolean(isOpen),
+            'mm-popover--reference-hidden': Boolean(referenceHidden),
+          },
+          className,
+        )}
+        ref={setPopperElement}
+        {...attributes.popper}
+        {...props}
+        style={{ ...styles.popper, ...contentStyle, ...props.style }}
+      >
+        {children}
+        {hasArrow && (
+          <Box
+            borderColor={BorderColor.borderMuted}
+            className={classnames('mm-popover__arrow')}
+            ref={setArrowElement}
+            display={DISPLAY.FLEX}
+            justifyContent={JustifyContent.center}
+            alignItems={AlignItems.center}
+            style={styles.arrow}
+            {...attributes.arrow}
+            {...arrowProps}
+          />
+        )}
+      </Box>
+    );
+
+    return (
+      <>
+        {isPortal
+          ? isOpen && createPortal(PopoverContent, document.body)
+          : isOpen && PopoverContent}
```

Comment:
Is the difference here between a Tooltip and a Menu? Where the menu component we would want to render inside the body near the anchor so it is tab-able using a keyboard and tooltip so it renders near the bottom of the body so it can remain on top of other elements?